### DISCUSSION TITLE
proxyMap/proxySet performance

### DISCUSSION THREAD

**Discussion by farwayer** [pmndrs/valtio#686] [General]
Hello. valtio looks interesting, thanks!

I am now figuring out how the library is suitable for serious development. And I was unpleasantly surprised when I saw the `proxySet/proxyMap` implementations.

An array is used as a data structure, and `.find` (and similar) is used to implement methods. Performance on large amounts of data will be terrible. And this is exactly where `Map` is usually used for speed/memory optimization.

I would advise changing the implementation or, if it is not possible to achieve performance comparable to `Set / Map`, it is better to remove these methods. Do not mislead library users who expect good performance.

**Comment by dai-shi** [comment]
See also #485.

Providing `Set / Map` impls for performance is out of the scope of this library.
Please consider developing a third-party library for that.
Please feel free to open a PR to fix docs to avoid misleading.

**Reply by farwayer** [reply]
Ah sorry missed #485 discussion. Thanks.
